pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -1176,84 +1188,1309 @@ func Test_server_Insert(t *testing.T) {
+			id2 := "uuid2"
+
+			return test{
+				name: "Decision Table Testing case 2.2: Insert duplicated request success when SkipStrictExistCheck is true (duplicated vector)",
+				args: args{
+					ctx: ctx,
+					req: &payload.Insert_Request{
+						Vector: &payload.Object_Vector{
+							Id:     id,
+							Vector: intVec,
+						},
+						Config: &payload.Insert_Config{
+							SkipStrictExistCheck: true,
+						},
+					},
+				},
+				fields: fields{
+					name: name,
+					ip:   ip,
+					svcCfg: &config.NGT{
+						Dimension:    intVecDim,
+						DistanceType: ngt.Angle.String(),
+						ObjectType:   ngt.Uint8.String(),
+						KVSDB:        &config.KVSDB{},
+						VQueue:       &config.VQueue{},
+					},
+					svcOpts: []service.Option{
+						service.WithEnableInMemoryMode(true),
+					},
+				},
+				beforeFunc: func(s *server) {
+					s.ngt.Insert(id2, intVec)
+				},
+				want: want{
+					wantRes: &payload.Object_Location{
+						Name: name,
+						Uuid: id,
+						Ips:  []string{ip},
+					},
+				},
+			}
+		}(),
+		func() test {
+			req := &payload.Insert_Request{
+				Vector: &payload.Object_Vector{
+					Id:     id,
+					Vector: intVec,
+				},
+				Config: &payload.Insert_Config{
+					SkipStrictExistCheck: true,
+				},
+			}
+
+			return test{
+				name: "Decision Table Testing case 2.3: Insert duplicated request fail when SkipStrictExistCheck is true (duplicated ID & vector)",
+				args: args{
+					ctx: ctx,
+					req: req,
+				},
+				fields: fields{
+					name: name,
+					ip:   ip,
+					svcCfg: &config.NGT{
+						Dimension:    intVecDim,
+						DistanceType: ngt.Angle.String(),
+						ObjectType:   ngt.Uint8.String(),
+						KVSDB:        &config.KVSDB{},
+						VQueue:       &config.VQueue{},
+					},
+					svcOpts: []service.Option{
+						service.WithEnableInMemoryMode(true),
+					},
+				},
+				beforeFunc: func(s *server) {
+					s.ngt.Insert(id, intVec)
+				},
+				want: want{
+					err: status.WrapWithAlreadyExists(fmt.Sprintf("Insert API uuid %s already exists", id), errors.ErrUUIDAlreadyExists(id),
+						&errdetails.RequestInfo{
+							RequestId:   req.GetVector().GetId(),
+							ServingData: errdetails.Serialize(req),
+						},
+						&errdetails.ResourceInfo{
+							ResourceType: ngtResourceType + "/ngt.Insert",
+							ResourceName: fmt.Sprintf("%s: %s(%s)", apiName, name, ip),
+						}),
+				},
+			}
+		}(),
 	}
 
 	for _, tc := range tests {
 		test := tc
 		t.Run(test.name, func(tt *testing.T) {
 			tt.Parallel()
 			defer goleak.VerifyNone(tt, goleak.IgnoreCurrent())
-			if test.beforeFunc != nil {
-				test.beforeFunc(test.args)
-			}
 			if test.afterFunc != nil {
 				defer test.afterFunc(test.args)
 			}
 			checkFunc := test.checkFunc
 			if test.checkFunc == nil {
 				checkFunc = defaultCheckFunc
 			}
+
+			eg, _ := errgroup.New(ctx)
+			ngt, err := service.New(test.fields.svcCfg, append(test.fields.svcOpts, service.WithErrGroup(eg))...)
+			if err != nil {
+				t.Error(err)
```

Comment:
I think you should unify the notation here with the others and use `tt`.
```suggestion
				tt.Errorf("error = %v", err)
```